### ISSUE TITLE
HIP Backend - ROCm 5.3 Warning & Error

### ISSUE THREAD

**Issue by kiritigowda** [GPUOpen-ProfessionalCompute-Libraries/MIVisionX#961]
Snap shot of error & warnings on ROCm 5.3
* Distro
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=22.04
DISTRIB_CODENAME=jammy
DISTRIB_DESCRIPTION="Ubuntu 22.04 LTS"
```
* Package Info
```
ii  rocm-clang-ocl                         0.5.0.50300-45~22.04                    amd64        OpenCL compilation with clang compiler.
ii  rocm-cmake                             0.8.0.50300-45~22.04                    amd64        rocm-cmake built using CMake
ii  rocm-core                              5.3.0.50300-45~22.04                    amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-dbgapi                            0.67.0.50300-45~22.04                   amd64        Library to provide AMD GPU debugger API
ii  rocm-debug-agent                       2.0.3.50300-45~22.04                    amd64        Radeon Open Compute Debug Agent (ROCdebug-agent)
ii  rocm-dev                               5.3.0.50300-45~22.04                    amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-device-libs                       1.0.0.50300-45~22.04                    amd64        Radeon Open Compute - device libraries
ii  rocm-gdb                               12.1.50300-45~22.04                     amd64        ROCgdb
ii  rocm-libs                              5.3.0.50300-45~22.04                    amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-llvm                              15.0.0.22362.50300-45~22.04             amd64        ROCm compiler
ii  rocm-llvm-alt                          15.0.0.22335.50300-45~22.04             amd64        Proprietary ROCm compiler
ii  rocm-ocl-icd                           2.0.0.50300-45~22.04                    amd64        opencl built using CMake
ii  rocm-opencl                            2.0.0.50300-45~22.04                    amd64        opencl built using CMake
ii  rocm-opencl-dev                        2.0.0.50300-45~22.04                    amd64        opencl built using CMake
ii  rocm-smi-lib                           5.0.0.50300-45~22.04                    amd64        AMD System Management libraries
ii  rocm-utils                             5.3.0.50300-45~22.04                    amd64        Radeon Open Compute (ROCm) Runtime software stack
ii  rocm-validation-suite                  1.0.50300.50300-45~22.04                amd64        ROCm Validation Suite validates the AMD platform using ROCm Runtime - The ROCm Validation
```
* Error
```
In file included from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/rocAL/rocAL/source/device/device_manager_hip.cpp:27:
/long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/rocAL/rocAL/./include/device/device_manager_hip.h:56:24: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
   56 | using pRocalHip = std::shared_ptr<DeviceManagerHip>;
      |                        ^~~~~~~~~~
/long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/rocAL/rocAL/./include/device/device_manager_hip.h:28:1: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   27 | #include "hip/hip_runtime.h"
  +++ |+#include <memory>
   28 | #include <vx_ext_amd.h>
make[2]: *** [rocAL/rocAL/CMakeFiles/rocal.dir/build.make:780: rocAL/rocAL/CMakeFiles/rocal.dir/source/device/device_manager_hip.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:734: rocAL/rocAL/CMakeFiles/rocal.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

```
* Warnings
```
[ 44%] Building CXX object amd_openvx_extensions/amd_rpp/CMakeFiles/vx_rpp.dir/source/internal_publishKernels.cpp.o
In file included from /opt/rocm/include/rpp/rpp.h:40,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_rpp.h:31,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_publishKernels.h:32,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/source/internal_publishKernels.cpp:23:
/opt/rocm-5.3.0/hip/include/hip/hip_runtime_api.h:25:113: note: ‘#pragma message: This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip’
   25 | #pragma message("This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip")
      |                                                                                                                 ^
In file included from /opt/rocm-5.3.0/include/hip/hip_runtime_api.h:35,
                 from /opt/rocm-5.3.0/hip/include/hip/hip_runtime_api.h:26,
                 from /opt/rocm/include/rpp/rpp.h:40,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_rpp.h:31,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_publishKernels.h:32,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/source/internal_publishKernels.cpp:23:
/opt/rocm-5.3.0/hip/include/hip/hip_version.h:25:113: note: ‘#pragma message: This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip’
   25 | #pragma message("This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip")
      |                                                                                                                 ^
In file included from /opt/rocm-5.3.0/include/hip/hip_runtime_api.h:36,
                 from /opt/rocm-5.3.0/hip/include/hip/hip_runtime_api.h:26,
                 from /opt/rocm/include/rpp/rpp.h:40,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_rpp.h:31,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_publishKernels.h:32,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/source/internal_publishKernels.cpp:23:
/opt/rocm-5.3.0/hip/include/hip/hip_common.h:25:113: note: ‘#pragma message: This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip’
   25 | #pragma message("This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip")
      |                                                                                                                 ^
In file included from /opt/rocm-5.3.0/include/hip/hip_runtime_api.h:480,
                 from /opt/rocm-5.3.0/hip/include/hip/hip_runtime_api.h:26,
                 from /opt/rocm/include/rpp/rpp.h:40,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_rpp.h:31,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_publishKernels.h:32,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/source/internal_publishKernels.cpp:23:
/opt/rocm-5.3.0/hip/include/hip/amd_detail/host_defines.h:25:113: note: ‘#pragma message: This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip’
   25 | #pragma message("This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip")
      |                                                                                                                 ^
In file included from /opt/rocm-5.3.0/include/hip/hip_runtime_api.h:481,
                 from /opt/rocm-5.3.0/hip/include/hip/hip_runtime_api.h:26,
                 from /opt/rocm/include/rpp/rpp.h:40,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_rpp.h:31,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/include/internal_publishKernels.h:32,
                 from /long_pathname_so_that_rpms_can_package_the_debug_info/data/driver/kiriti/MIVisionX/amd_openvx_extensions/amd_rpp/source/internal_publishKernels.cpp:23:
/opt/rocm-5.3.0/hip/include/hip/driver_types.h:25:113: note: ‘#pragma message: This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip’
   25 | #pragma message("This file is deprecated. Use file from include path /opt/rocm-ver/include/ and prefix with hip")
      |                                                            
```

**Comment by kiritigowda** [comment]
@asalmanp  -- Docker to use `compute-artifactory.amd.com:5000/rocm-plus-docker/compute-rocm-rel-5.3:45-ubuntu-22.04-stg2 `

**Comment by kiritigowda** [comment]
HIP Message -- https://github.com/ROCm-Developer-Tools/hipamd/blob/9ce0cf841d512c6a2fa88616b47e3a4d59338ac5/hip-config.cmake.in#L232

**Comment by asalmanp** [comment]
@kiritigowda [ROCm5.3](https://docs.amd.com/bundle/ROCm-Installation-Guide-v5.3/page/How_to_Install_ROCm.html) is now public. I tested with the public version on UB22.04, and I am not able to reproduce the warning related to the hip mentioned in this issue. Please test with the public version and let me know if you still see the issue. Thanks.

**Comment by kiritigowda** [comment]
@asalmanp -- seeing this issue on ubuntu 22.04 base machine

```
In file included from /home/kiriti/develop/mivisionx-kiriti/MIVisionX/rocAL/rocAL/source/device/device_manager_hip.cpp:27:
/home/kiriti/develop/mivisionx-kiriti/MIVisionX/rocAL/rocAL/./include/device/device_manager_hip.h:56:24: error: ‘shared_ptr’ in namespace ‘std’ does not name a template type
   56 | using pRocalHip = std::shared_ptr<DeviceManagerHip>;
      |                        ^~~~~~~~~~
/home/kiriti/develop/mivisionx-kiriti/MIVisionX/rocAL/rocAL/./include/device/device_manager_hip.h:28:1: note: ‘std::shared_ptr’ is defined in header ‘<memory>’; did you forget to ‘#include <memory>’?
   27 | #include "hip/hip_runtime.h"
  +++ |+#include <memory>
   28 | #include <vx_ext_amd.h>
[ 78%] Building CXX object rocAL/rocAL/CMakeFiles/rocal.dir/source/device/ocl_setup.cpp.o
[ 78%] Building CXX object rocAL/rocAL/CMakeFiles/rocal.dir/source/loaders/circular_buffer.cpp.o
[ 79%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_siftCompute.cpp.o
[ 79%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_siftDetect.cpp.o
make[2]: *** [rocAL/rocAL/CMakeFiles/rocal.dir/build.make:780: rocAL/rocAL/CMakeFiles/rocal.dir/source/device/device_manager_hip.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 79%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_simpleBlobDetector.cpp.o
[ 79%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_sobel.cpp.o
[ 80%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_starFeatureDetector.cpp.o
[ 80%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_subtract.cpp.o
[ 80%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_surfCompute.cpp.o
[ 80%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_surfDetect.cpp.o
[ 81%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_threshold.cpp.o
[ 81%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_transpose.cpp.o
[ 81%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_warpAffine.cpp.o
[ 81%] Building CXX object amd_openvx_extensions/amd_opencv/CMakeFiles/vx_opencv.dir/source/OpenCV_warpPerspective.cpp.o
make[1]: *** [CMakeFiles/Makefile2:734: rocAL/rocAL/CMakeFiles/rocal.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 82%] Linking CXX shared library ../../lib/libvx_opencv.so
[ 82%] Built target vx_opencv
make: *** [Makefile:156: all] Error 2
```

**Comment by kiritigowda** [comment]
@asalmanp the warnings are fixed on docker by checking for ROCM_PATH in the env variable -- https://github.com/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/blob/7575e4edaf3991e7a7cbf31973c15b0e59b0f4a4/MIVisionX-setup.py#L73

**Comment by kiritigowda** [comment]
Develop branch has the fix, which will be merged to master with #957
